Polygon.js

```diff
@@ -0,0 +1,185 @@
+
+    Copyright © 2015 czar
+    http://en.wikipedia.org/wiki/User_talk:Czar
+
+    This file is part of Zotero.
+
+    Zotero is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Zotero is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+    ***** END LICENSE BLOCK *****
+*/
+
+function pageType (doc, url) { 
+	if (temp = ZU.xpathText(doc, '//script[contains(text(), "page_type :")]'))
+		{ return temp.match(/page_type\s?:\s?"([^"]*)"/)[1]; }
+	else return null;
+}
+
+function detectWeb(doc, url) {
+	switch (pageType(doc, url)) { 
+		case "Article":
+			return "webpage";
+		case "HubPage":
+			return "multiple";
+	}
+	if (url.indexOf("/search?q=")!=-1){ return "multiple" }
+	else return null;
+}
+
+function scrape(doc, url) {
+	var newItem = new Zotero.Item("webpage");
+	newItem.websiteTitle = "Polygon";
+	newItem.language = "en-US";
+
+	newItem.url = url;
+	newItem.title = ZU.xpathText(doc, '//h2[@class="m-entry__title"]');
+	newItem.date = ZU.xpathText(doc, '//meta[@property="article:published_time"]/@content').split("T")[0];
+
+	newItem.abstractNote = ZU.xpathText(doc, '//meta[@name="description"]/@content');
+	newItem.attachments.push({document:doc, title:"Polygon snapshot" ,mimeType:"text/html"});
+
+		// Authors
+	if ( authorMetadata = ZU.xpathText(doc, '//script[contains(text(), "dataLayer =")]') ) {
+		authorMetadata = authorMetadata.match(/"Author"\s?:\s?"([^"]*)"/)[1].split(":");
+		do {
+			newItem.creators.push(ZU.cleanAuthor(authorMetadata[0], "author"));
+			authorMetadata.shift();
+		}
+		while (authorMetadata.length);
+	}
+	
+	newItem.complete();
+}
+
+function doWeb(doc, url) {
+	switch (detectWeb(doc, url)) { 
+		case "webpage":
+			return scrape(doc, url);
+		case "multiple":
+			var items = {};
+	        articles = [];
+	        var titles = ZU.xpath(doc, '//*[contains(concat(" ", normalize-space(@class), " "), "m-block__meta")]/h2/a') //this is needed to catch the class attribute when among others, e.g., "m-block__meta meta" on search pages
+	        for (var i = 0; i < titles.length; i++) {
+	            items[titles[i].href] = titles[i].textContent;
+	        }
+	        Zotero.selectItems(items, function(items) {
+	            if (!items) {
+	                return true;
+	            }
+	            for (var i in items) {
+	                articles.push(i);
+	            }
+	            Zotero.Utilities.processDocuments(articles, scrape);
+	        })
+		default:
```

Comment:
right, but the only way you can run `doWeb` when `detectWeb` returns nothing is in Scaffold (and there, you can also run it on a site outside of the target regex. Like you could run it here on github, in which case you'd also get an error). In Zotero itself, doWeb only becomes available after detectWeb returns a legal itemType. 
And I'm assuming Citoid, too, will stop as soon as detectWeb fails (otherwise it'd be running into way more problems).